validphys2/src/validphys/scripts/get_hepdata.py

```diff
@@ -0,0 +1,187 @@
+from reportengine.checks import CheckError
+
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.INFO)
+log.addHandler(colors.ColorHandler())
+
+
+class DownloadFail(Exception):
+    pass
+
+
+class VersionMismatch(Exception):
+    pass
+
+
+class HepDataConfig:
+    
+    def __init__(self, metadata_yaml: str, path: Path=None, force: bool=False) -> None:
+        """
+        Parameters
+        ----------
+            metadata_yaml: str
+                Path to the metadata YAML file
+            path: Path
+                Path to create the folder in which the tables will be downloaded
+            force: bool
+                Decide on whether or not overwrite the existing tables
+        """
+
+        hep_metadata = self.extract_metadata(metadata_yaml)
+        self.url = hep_metadata["url"]
+        self.version = int(hep_metadata["version"])
+        self.force_download = force
+        self.tables = hep_metadata["tables"]
+
+        # Instantiate the dictionary of the scrapped url
+        self.hep_webinfo = requests.get(self.url).json()
+        self.webversion = self.hep_webinfo["version"]
+        self.tables_dict = self.hep_webinfo["hasPart"]
+        self.folder = Path().absolute().joinpath("rawdata") if path is None else path
+
+        # Check at initialisation time if local & web versions match. The following is
+        # also used to check if the already downloaded HepData tables are outdated. This
+        # however assumes any existing tables were downloaded with the version found in
+        # the metadata.yaml file (as it should be).
+        if self.version != self.webversion and not self.force_download:
+            sys.exit(
+                f"The required version and the HepData one do not match. The version in the "
+                f"metadata is v{self.version} while the one on HepData is v{self.webversion}. "
+                f"Please use the flag --force in order to overwrite tables."
+            )
+
+    @staticmethod
+    def extract_metadata(metadata_path: str) -> dict:
+        """
+        Extract specific information from the metadata, specifically the HepData
+        URL and version with the corresponding table number.
+
+        metadata_path: str
+            Path to the metadata YAML file
+        """
+        with open(metadata_path, 'r') as stream_metadata:
+            metadata_dic = yaml.safe_load(stream_metadata)
+        return metadata_dic["hepdata"]
+
+    def write_tables(self, file: bytes, table_id: Optional[int] = None) -> None:
+        """
+        Write a given table into a file and save it into the disk afterwards.
+
+        Parameters
+        ----------
+            file: bytes
+                HepData table in bytes
+            table_id: int
+                Denote the `table_id`-th table
+        """
+        self.folder.mkdir(exist_ok=True)
+        ins_id = self.url.split("/")[-1]
+        filename = f"HEPData-{ins_id}-v{self.version}-Table_{table_id}"
```

Comment:
It doesn't look like table_id=None works particularly well here. Why is it in the function signature?